src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -25,6 +25,7 @@
     Optional,
     Tuple,
     Type,
+    TypedDict,
```

Comment:
TypedDict and total arguments are supported starting with Python 3.8.

https://docs.python.org/3.8/library/typing.html?highlight=typeddict#typing.TypedDict